app/src/main/java/com/nativework/covid19vaccinetracker/ui/HomeActivity.kt

```diff
@@ -0,0 +1,183 @@
+package com.nativework.covid19vaccinetracker.ui
+
+import android.content.Intent
+import android.os.Bundle
+import android.util.Log
+import android.view.View
+import android.widget.Toast
+import androidx.lifecycle.ViewModelProvider
+import com.nativework.covid19vaccinetracker.R
+import com.nativework.covid19vaccinetracker.base.BaseApp
+import com.nativework.covid19vaccinetracker.databinding.ActivityHomeBinding
+import com.nativework.covid19vaccinetracker.models.locality.District
+import com.nativework.covid19vaccinetracker.models.locality.StatesList
+import com.nativework.covid19vaccinetracker.ui.center.CenterActivity
+import com.nativework.covid19vaccinetracker.ui.home.AutocompleteAdapter
+import com.nativework.covid19vaccinetracker.utils.AppUtils
+import com.nativework.covid19vaccinetracker.utils.Constants
+import java.text.SimpleDateFormat
+import java.util.*
+import kotlin.collections.ArrayList
+
+class HomeActivity : BaseApp() {
+
+    private lateinit var binding: ActivityHomeBinding
+    private var viewModel: HomeViewModel? = null
+    private var adapter: AutocompleteAdapter? = null
+    private var stateStringList = ArrayList<String>()
+    private var statesList: StatesList? = null
+    private var districtList = ArrayList<District>()
+    private var districtStringList = ArrayList<String>()
+    private var districtId: String? = null
+    private var dateSelected: String? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityHomeBinding.inflate(layoutInflater)
+        val view = binding.root
+        setContentView(view)
+        title = "Search Vaccine Centers"
+        viewModel = ViewModelProvider(this).get(HomeViewModel::class.java)
+        setListeners()
+        initData()
+        setObservers()
+    }
+
+    private fun setListeners() {
+        binding.autoTextState.setOnItemClickListener { adapterView, _, position, _ ->
+            Toast.makeText(
+                this,
+                adapterView.getItemAtPosition(position) as String,
+                Toast.LENGTH_SHORT
+            ).show()
+            val itemSelected = adapterView.getItemAtPosition(position)
+            var stateId = ""
+            for (state in statesList!!) {
+                if (itemSelected.equals(state.stateName)) {
+                    stateId = state.stateId.toString()
+                }
+            }
+            Log.d("tag", stateId)
```

Comment:
Not sure if we are keeping logs or to use Logger for better removal of logs in later stages